schematools/permissions.py

```diff
@@ -46,12 +46,20 @@ def revoke_permissions(engine, role):
 def apply_schema_and_profile_permissions(
     engine, ams_schema, profiles, role, scope, dry_run=False, create_roles=False
 ):
-    if ams_schema:
-        create_acl_from_schemas(engine, ams_schema, role, scope, dry_run, create_roles)
-    if profiles:
-        profile_list = profiles.values()
-        create_acl_from_profiles(engine, "public", profile_list, role, scope)
-
+    Session = sessionmaker(bind=engine)
```

Comment:
Goed om het in een transactie te doen! Ik vroeg me nog af waarom je de session gebruikt, omdat die toch vooral bij de ORM van SQLAlchemy hoort, terwijl er alleen raw sql wordt gebruikt. Je zou dan ook dit kunnen doen:
```
with engine.begin() as connection:
    connection.execute(...)  # of evt de connection doorgeven
```
Deze context processor doet automatisch een transactie.

Reply:
Klopt, ik twijfelde. Het doorgeven van een connection naar een andere functie, die hem dan weer verder kan doorgeven, is blijkbaar een anti pattern, zie https://docs.sqlalchemy.org/en/13/core/connections.html en dan vooral het stukje bij Arbitrary Transaction Nesting as an Antipattern. De Session uit de ORM verzorgt context management, en dan is dat blijkbaar geen issue.